src/TDB2.cpp

```diff
@@ -57,6 +57,15 @@ TDB2::TDB2 ()
 ////////////////////////////////////////////////////////////////////////////////
 void TDB2::open_replica (const std::string& location, bool create_if_missing)
 {
+  File pending_data = File (location + "/pending.data");
+  if (pending_data.exists()) {
+    Color warning = Color (Context::getContext ().config.get ("color.warning"));
+    std::cout << warning.colorize (
+      format ("Found existing '.data' files in {1}", location)) << "\n";
+    std::cout << "  Taskwarrior's storage format changed in 3.0.\n";
```

Comment:
I'm not in love with this wording but I feel like some kind of "call to action" other than (implicitly) "delete your .data files" would be good.

```suggestion
    std::cout << "  Taskwarrior's storage format changed in 3.0, requiring a manual migration.\n";
```